app/src/main/java/com/ranveeraggarwal/letrack/activities/AddPersonActivity.java

```diff
@@ -37,24 +48,48 @@
     String selectedFrequency;
     String selectedStartDate;
 
+    int REQUEST_CONTACTS = 1;
+    String TAG = "AddPersonActivity";
+    boolean hasContactPermission = false;
+
     DatabaseAdapter databaseAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setContentView(R.layout.activity_add_person);
+        if (ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.READ_CONTACTS}, REQUEST_CONTACTS);
+        } else {
+            hasContactPermission = true;
+        }
```

Comment:
I haven't used permissions before, isn't it possible to get that dialog back once the permission has been denied?
Because if it is, we can directly trigger that dialog if the user denies the first time. It isn't wrong to do that because the user knows that he/she needs to give permissions to access contacts to use that feature.